lib/chefspec/server_runner.rb

```diff
@@ -99,5 +99,24 @@ def port
 
       return @port
     end
+
+    #
+    # The path to cache files on disk.  Unlike the SoloRunner, this path needs
+    # to remain consistent for every Chef run.  Else the Chef client tries to
+    # loads the same cookbook multiple times and will encounter deprecated logic
+    # when creating LWRPs.  This method can be removed when the deprecated logic
+    # is removed.
+    #
+    # The method adds a {Kernel.at_exit} handler to ensure the temporary
+    # directory is deleted when the system exits.
+    #
+    def file_cache_path
+      @@file_cache_path ||= begin
```

Comment:
Is there a reason this needs to be a class variable? This isn't threadsafe and has race conditions as written.